src/Arcus.Messaging.Abstractions/MessageContext.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System.Collections.Generic;
+using GuardNet;
+
+namespace Arcus.Messaging.Abstractions
+{
+    /// <summary>
+    ///     Contextual information concerning a message
+    /// </summary>
+    public class MessageContext
+    {
+        /// <summary>
+        ///     Unique identifier of the message
+        /// </summary>
+        public string MessageId { get; }
+
+        /// <summary>
+        ///     Contextual properties provided on the message
+        /// </summary>
+        public IDictionary<string, object> Properties { get; }
+
+        /// <summary>
+        ///     Constructor
+        /// </summary>
+        /// <param name="messageId">Unique identifier of the message</param>
+        /// <param name="properties">Contextual properties provided on the message</param>
+        public MessageContext(string messageId, IDictionary<string, object> properties)
+        {
+            Guard.NotNullOrEmpty(messageId, nameof(messageId));
+            Guard.NotNull(properties, nameof(properties));
```

Comment:
Maybe also verify each key/value on each property?

Reply:
What are you thinking of exactly?

Reply:
That each key is not empty and each value isn't `null`.

Reply:
Is that up to us to determine what is considered an invalid message context entry?

Personally I think we should take whatever the depending queueing system gives us and pass it as is, no?

Reply:
Ah, yes; It's more of a DTO type here. OK, is good.